tests/e2e/test_spyre_cb.py

```diff
@@ -83,177 +91,415 @@ def test_cb_handling(
+    request_outputs = engine.step()
+    assert len(request_outputs) == 1
+    assert request_outputs[0].request_id == "3"  # req 3 is decoding (prefill)
+
+    assert len(engine_core.scheduler.waiting) == 0
+    assert len(engine_core.scheduler.running) == 2
+
+    request_outputs = engine.step()
+    assert len(request_outputs) == 2  # requests 2 and 3 decoding now
+    assert request_outputs[0].request_id == "3"  # req 3 is decoding
+    assert request_outputs[1].request_id == "2"  # req 2 is decoding
+
+    assert len(engine_core.scheduler.waiting) == 0
+    assert len(engine_core.scheduler.running) == 2
+    request_outputs = engine.step()
+    assert len(request_outputs) == 2  # requests 2 and 3 decoding now
+    assert request_outputs[0].request_id == "3"  # req 3 is decoding
+    assert request_outputs[1].request_id == "2"  # req 2 is decoding
+
+    assert len(engine_core.scheduler.waiting) == 0
+    assert len(engine_core.scheduler.running) == 2
+    request_outputs = engine.step()
+    assert len(request_outputs) == 2  # requests 2 and 3 decoding now
+    assert request_outputs[0].request_id == "3"  # req 3 is decoding
+    assert request_outputs[1].request_id == "2"  # req 2 is decoding
+
+    assert len(engine_core.scheduler.waiting) == 0
+    assert len(engine_core.scheduler.running) == 2
+    request_outputs = engine.step()
+    assert len(request_outputs) == 2  # requests 2 and 3 decoding now
+    assert request_outputs[0].request_id == "3"  # req 3 is decoding
+    assert request_outputs[1].request_id == "2"  # req 2 is decoding
+
+    assert len(engine_core.scheduler.waiting) == 0
+    assert len(engine_core.scheduler.running) == 2
+    request_outputs = engine.step()
+    assert len(request_outputs) == 2  # requests 2 and 3 decoding now
+    assert request_outputs[0].request_id == "3"  # req 3 is decoding
+    assert request_outputs[1].request_id == "2"  # req 2 is decoding
+
+    assert len(engine_core.scheduler.waiting) == 0
+    assert len(engine_core.scheduler.running) == 2
+    request_outputs = engine.step()
+    assert len(request_outputs) == 2  # requests 2 and 3 decoding now
+    assert request_outputs[0].request_id == "3"  # req 3 is decoding
+    assert request_outputs[1].request_id == "2"  # req 2 is decoding
+    assert request_outputs[1].finished  # request 2 is done
+    assert request_outputs[1].outputs[0].text == " 6 5 4 3 2 "
+
+    assert len(engine_core.scheduler.waiting) == 0
+    assert len(engine_core.scheduler.running) == 1
+
+
+def create_random_request(
+        request_id: int, num_tokens: int,
+        sampling_params: SamplingParams) -> EngineCoreRequest:
+    return EngineCoreRequest(
+        request_id=str(request_id),
+        prompt_token_ids=[request_id] * num_tokens,
+        mm_inputs=None,
+        mm_hashes=None,
+        mm_placeholders=None,
+        sampling_params=sampling_params,
+        eos_token_id=None,
+        arrival_time=0,
+        lora_request=None,
+    )
```

Comment:
we need to add `cache_salt=None` as part of `EngineCoreRequest` to make the `vllm:main` check pass

```suggestion
def create_random_request(
        request_id: int, num_tokens: int,
        sampling_params: SamplingParams) -> EngineCoreRequest:
    return EngineCoreRequest(
        request_id=str(request_id),
        prompt_token_ids=[request_id] * num_tokens,
        mm_inputs=None,
        mm_hashes=None,
        mm_placeholders=None,
        sampling_params=sampling_params,
        eos_token_id=None,
        arrival_time=0,
        lora_request=None,
        cache_salt=None
    )
```

Reply:
The cb test now fail with the default vllm version. We need to update it.

Reply:
oh whoops yeah good point